Kconfig.zephyr

```diff
@@ -442,6 +442,14 @@ config COMPILER_COLOR_DIAGNOSTICS
 	help
 	  Compiler diagnostic messages are colorized.
 
+config COMPILER_STATIC_ANALYSIS
+	bool "Compiler static analysis"
+	depends on "${ZEPHYR_TOOLCHAIN_VARIANT}" = "zephyr"
+	default n
+	help
+	  Enables an static analysis of program flow and issues
+	  warnings for problems found.
```

Comment:
There should be a note saying tha this only works with GCC for now, and enabling this option may significantly slow down the build process and can lead to an excessive number of false positive warnings.